### PR TITLE
Set progressbar refresh rate in Google Colab

### LINKED ISSUE #5515: Improve handling progressbar refresh rate in Google Colab
## 🚀 Feature

Set progress bar refresh rate automatically in Colab (as is right now), but still allow user to override if needed.

### Motivation

Want no warnings in colab, but at the same time not crash the browser and also not prevent user from overriding the value.

### Pitch

Before:
```python
Trainer(progress_bar_refresh_rate: int = 1)
```

Now:

Remove warning and implement the following:

```python
Trainer(progress_bar_refresh_rate: int = None)

if colab and refresh_rate = None:
    refresh_rate = 20
elif refresh_rate = None
   refresh_rate = 1

# otherwise we keep value passed in by user
```

### Alternatives

Currently, user can't override the value in Colab. 

### Additional context

Discussed in slack with @edenlightning

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #5515

Colab example based on this branch:
https://colab.research.google.com/drive/1CX9WOkZPB5HtqOQD0_rHPeaOa9Cca-1g?usp=sharing

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

### CODE REVIEW

**Comment by rohitgr7** [comment]
so refresh_rate is 1 now just because of the colab thing? The warning is raised during Trainer init itself, so isn't it better than setting it to 1 in other cases?

**Comment by awaelchli** [comment]
> so refresh_rate is 1 now just because of the colab thing?

no, the other way around. It is always one (as it was before) _except_ when in colab and except the user sets it to the value they want. 
I was requested to remove the warning

**Comment by rohitgr7** [comment]
> > so refresh_rate is 1 now just because of the colab thing?
> 
> no, the other way around. It is always one (as it was before) _except_ when in colab and except the user sets it to the value they want.
> I was requested to remove the warning

ok my bad... got confused it with `log_every_n_steps`.

**Review by rohitgr7 — APPROVED** [review]

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ def __init__(
             process_position: orders the progress bar when running multiple models on same machine.
 
             progress_bar_refresh_rate: How often to refresh progress bar (in steps). Value ``0`` disables progress bar.
-                Ignored when a custom callback is passed to :paramref:`~Trainer.callbacks`.
+                Ignored when a custom callback is passed to :paramref:`~Trainer.callbacks`. Default: 1
```
Comment (SkafteNicki) [inline pytorch_lightning/trainer/trainer.py:222]:
```suggestion
                Ignored when a custom progress bar is passed to :paramref:`~Trainer.callbacks`. Default: 1
```

**Review by SkafteNicki — APPROVED** [review]

**Review by ydcjeff — APPROVED** [review]

File: pytorch_lightning/trainer/trainer.py [hunk 2]
```diff
@@ -219,7 +219,7 @@ def __init__(
             process_position: orders the progress bar when running multiple models on same machine.
 
             progress_bar_refresh_rate: How often to refresh progress bar (in steps). Value ``0`` disables progress bar.
-                Ignored when a custom callback is passed to :paramref:`~Trainer.callbacks`.
+                Ignored when a custom progress bar is passed to :paramref:`~Trainer.callbacks`. Default: 1
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer.py:222]: this is confusing as you say default but in code is none, so maybe tell that it is set internally?

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/trainer/trainer.py (hunk 2 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/trainer.py:222]: what should I do?
we have other trainer args set as None and they are internally set to the defaults

File: pytorch_lightning/trainer/connectors/callback_connector.py [hunk 3]
```diff
@@ -75,14 +75,11 @@ def configure_checkpoint_callbacks(self, checkpoint_callback: Union[ModelCheckpo
...
-    def configure_progress_bar(self, refresh_rate=1, process_position=0):
-        # smaller refresh rate on colab causes crashes, warn user about this
-        if os.getenv('COLAB_GPU') and refresh_rate < 20:
-            rank_zero_warn(
-                "You have set progress_bar_refresh_rate < 20 on Google Colab. This"
-                " may crash. Consider using progress_bar_refresh_rate >= 20 in Trainer.",
-                UserWarning
-            )
+    def configure_progress_bar(self, refresh_rate=None, process_position=0):
+        if os.getenv('COLAB_GPU') and refresh_rate is None:
```
Comment (justusschock) [inline pytorch_lightning/trainer/connectors/callback_connector.py:79]: is this env always set for colab? also on TPU?

File: pytorch_lightning/trainer/trainer.py (hunk 2 above)
Comment (justusschock) [reply pytorch_lightning/trainer/trainer.py:222]: yes, but usually this is only done for one of the following reasons:

1.) We need to check for some type before setting the default (for example to raise deprecation warnings)
2.) They are mutable otherwise.
3.) They depend on other parameters
4.) They are really hard to parse/set correctly

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/trainer/trainer.py (hunk 2 above)
Comment (Borda) [reply pytorch_lightning/trainer/trainer.py:222]: In general the 2) is most common

File: pytorch_lightning/trainer/connectors/callback_connector.py (hunk 3 above)
Comment (rohitgr7) [reply pytorch_lightning/trainer/connectors/callback_connector.py:79]: it's `'0'` otherwise

Comment (awaelchli) [reply pytorch_lightning/trainer/connectors/callback_connector.py:79]: @rohitgr7 not according to my quick testing.
https://colab.research.google.com/drive/1_W9Dk-cxbBZmg6R2tNrqu3gzpjcEPjdb?usp=sharing
I get True back on CPU, GPU, and TPU even after restarting the runtime.

File: pytorch_lightning/trainer/trainer.py (hunk 2 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/trainer.py:222]: Ok, here we are in case 3, because the default depends on an environment variable.
Then I will just update the docstring, or do you request other changes?

Comment (awaelchli) [reply pytorch_lightning/trainer/trainer.py]: updated docstring, do you like it better?

**Review by tchaton — APPROVED** [review]
LGTM !
